src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -27,25 +27,15 @@ public AuthorizationCheckService(UserService userService) {
     }
 
     public <T extends Exercise> boolean isAtLeastTeachingAssistantForExercise(Optional<T> exercise) {
-        if (exercise.isPresent()) {
-            return isAtLeastTeachingAssistantForExercise(exercise.get());
-        }
-        else {
-            return false;
-        }
+        return exercise.filter(this::isAtLeastTeachingAssistantForExercise).isPresent();
     }
 
     public boolean isAtLeastTeachingAssistantForExercise(Exercise exercise) {
         return isAtLeastTeachingAssistantInCourse(exercise.getCourse(), null);
     }
 
     public boolean isAtLeastTeachingAssistantForExercise(Optional<Exercise> exercise, User user) {
-        if (exercise.isPresent()) {
-            return isAtLeastTeachingAssistantForExercise(exercise.get(), user);
-        }
-        else {
-            return false;
-        }
+        return exercise.filter(exercise1 -> isAtLeastTeachingAssistantForExercise(exercise1, user)).isPresent();
```

Comment:
Are you 100% sure that the code does the same? What happens if the exercise is not present?
I don't like the variable name `exercise1` --> Please use a different one that is more expressive

Reply:
As above, `exercise` is Optional, so if it is not present, the `isPresent` will return false